src/app/icd/BUILD.gn

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+import("icd.gni")
+
+config("icd_server_config") {
+  defines = [ "CHIP_CONFIG_ICD_SERVER_ENABLE=${chip_enable_icd_server}" ]
+}
+
+source_set("icd-manager-srcs") {
+  sources = [
+    "${chip_root}/src/app/icd/ICDManager.cpp",
+    "${chip_root}/src/app/icd/ICDManager.h",
+  ]
+
+  public_deps = [ "${chip_root}/src/lib/core" ]
+}
+
+source_set("server-srcs") {
```

Comment:
Could you please document the difference between these two source sets?  Because "server" doesn't make much sense on its own: ICDManager is server-only anyway, as currently constituted, no?  But it's hard to say...

Fundamentally, we should have different headers/APIs for the "ICD client" and "ICD server" bits.

Reply:
Source set is split into two because the test BUILD.gn only requires the manager dependencies. 
This way, if extra stuff is needed for the ICDManager code, it also takes affect in the test build.gn.

I'll also add comments to clarify the actual source sets.

Reply:
done